src/features/smSystem/suppliersOrders/routers/OrderDetailsPage.tsx

```diff
@@ -0,0 +1,212 @@
+import { Button, CircularProgress, Stack, TextField, Typography } from '@mui/material';
+import { useState } from 'react';
+
+import { useGetOrderDetails } from '../api/useGetOrderDetails';
+import ProductDetailsInOrderTable from '../tables/ProductDetailsInOrderTable';
+import ProductsInOrderTable from '../tables/ProductsInOrderTable';
+
+const generateTxtFile = (content: string, fileName: string) => {
+  const blob = new Blob([content], { type: 'text/plain' });
+  const url = URL.createObjectURL(blob);
+  const link = document.createElement('a');
+  link.href = url;
+  link.download = fileName;
+  link.click();
+  URL.revokeObjectURL(url);
+};
+
+export const OrderDetailsPage = () => {
+  const getIdFromUrl = () => {
+    const url = window.location.hash;
+    const match = url.match(/orders\/(\d+)/);
+
+    if (match) {
+      const id = parseInt(match[1], 10);
+      return Number.isInteger(id) ? id : 0;
+    }
+    return 0;
+  };
+
+  const id = getIdFromUrl();
+  const { data, isLoading, isError } = useGetOrderDetails(id);
+
+  const [selectedProductId, setSelectedProductId] = useState<number>(0);
+
+  const [filterText, setFilterText] = useState('');
+
+  const productsInOrderTableData = data?.products_to_order.map(
+    ({ id, name, orders_per_branch }) => {
+      const totalToOrder = orders_per_branch.reduce(
+        (sum, order) => sum + order.to_order_amount,
+        0
+      );
+  
+      return {
+        id,
+        name,
+        totalToOrder,
+      };
+    }
+  );
```

Comment:
If it's needed, hmm you probably can do this per row in on table level, it's less heavy operation than calculating this value for every products. 
Otherwise, it's good practice to wrap it with `useMemo` if you want to read sth about it, but you can skip useMemo to just deliver it